File: apps/android/src/main/kotlin/com/points/android/ui/theme/PointHue.kt
package com.points.android.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import kotlin.math.abs

// Accent is NOT a fixed brand color. Every point type owns a hue;
// lightness & chroma are constant so all hues read at one weight.
//   light: L 60% · C 0.095     dark: L 70% · C 0.105
enum class PointHue(val degrees: Int, val light: Color, val dark: Color) {
    GREEN(152, Color(0xFF519164), Color(0xFF69B17E)),
    BLUE(215, Color(0xFF278EA4), Color(0xFF3EAFC7)),
    VIOLET(285, Color(0xFF7A78B7), Color(0xFF9895DD)),
    AMBER(40, Color(0xFFB06C54), Color(0xFFD6876C)),
    RED(18, Color(0xFFB2686B), Color(0xFFD88386)),
    MAGENTA(330, Color(0xFFA06B9B), Color(0xFFC386BE)),
    ;

    companion object {
        /** Maps a point type's stored hue (color-wheel degrees) to the nearest palette entry. */
        fun forDegrees(hue: Int): PointHue {
            val deg = ((hue % 360) + 360) % 360
            return entries.minBy { entry ->
                val d = abs(entry.degrees - deg)
                minOf(d, 360 - d)
            }
        }
    }
}

@Composable
fun PointHue.accent(): Color =
    if (isSystemInDarkTheme()) dark else light

File: apps/ios/Sources/Theme/PointHue.swift
import SwiftUI

// Accent is NOT a fixed brand color. Every point type owns a hue; lightness &
// chroma are constant so all hues read at one weight.
//   light: L 60% · C 0.095      dark: L 70% · C 0.105
//
// In SwiftUI this becomes `.tint(point.hue.color)`, set per tile — never a
// global accentColor. Color is reserved for data.
enum PointHue: String, CaseIterable, Codable {
    case green   // 152°
    case blue    // 215°
    case violet  // 285°
    case amber   // 40°
    case red     // 18°
    case magenta // 330°

    /// The color-wheel angle (degrees) each hue sits at — matches the Android palette and the design.
    var degrees: Int {
        switch self {
        case .green: return 152
        case .blue: return 215
        case .violet: return 285
        case .amber: return 40
        case .red: return 18
        case .magenta: return 330
        }
    }

    /// Maps a point type's stored hue (color-wheel degrees) to the nearest palette entry.
    static func forDegrees(_ hue: Int) -> PointHue {
        let deg = ((hue % 360) + 360) % 360
        return allCases.min { a, b in
            let da = min(abs(a.degrees - deg), 360 - abs(a.degrees - deg))
            let db = min(abs(b.degrees - deg), 360 - abs(b.degrees - deg))
            return da < db
        } ?? .green
    }

    var color: Color {
        switch self {
        case .green:   return Color(light: Color(red: 0.318, green: 0.569, blue: 0.392),
                                    dark:  Color(red: 0.412, green: 0.694, blue: 0.494))
        case .blue:    return Color(light: Color(red: 0.153, green: 0.557, blue: 0.643),
                                    dark:  Color(red: 0.243, green: 0.686, blue: 0.780))
        case .violet:  return Color(light: Color(red: 0.478, green: 0.471, blue: 0.718),
                                    dark:  Color(red: 0.596, green: 0.584, blue: 0.867))
        case .amber:   return Color(light: Color(red: 0.690, green: 0.424, blue: 0.329),
                                    dark:  Color(red: 0.839, green: 0.529, blue: 0.424))
        case .red:     return Color(light: Color(red: 0.698, green: 0.408, blue: 0.420),
                                    dark:  Color(red: 0.847, green: 0.514, blue: 0.525))
        case .magenta: return Color(light: Color(red: 0.627, green: 0.420, blue: 0.608),
                                    dark:  Color(red: 0.765, green: 0.525, blue: 0.745))
        }
    }
}
